Sources/SwiftLexicalLookup/SwiftLexicalLookup.docc/LookupRules.md

```diff
@@ -0,0 +1,78 @@
+# Unqualified Lookup Rules
+
+## Overview
+Unqualified lookup can be initiated from any source position, optionally targeting a specific identifier for name comparison. Unqualified lookup traverses the syntax tree from bottom-up, collecting names that match the lookup criteria. A name matches the lookup if the specified identifier (if present) refers to the introduced name and the position from which the unqualified lookup was triggered is preceded by the position where the name was introduced (if specified).
+
+## Default Scope Behavior
+Names are introduced and their availabilities are set by special syntax nodes referred to as syntax scopes or simply scopes. By default, each scope for lookup produces a result object that associates the scope of origin with the matched names. Lookup is then invoked on the parent scope, i.e., the closest syntax scope ancestor of the original scope in the syntax tree.
+
+## Name Kinds
+There are three distinct kinds of names identified by unqualified lookup:
+- **Identifier**: Represents `IdentifiableSyntax` syntax nodes introducing names such as `IdentifierPatternSyntax`, `ClosureParameterSyntax`, or `FunctionParameterSyntax`.
+- **Declaration**: Represents `NamedDeclSyntax` syntax nodes introducing names such as `ClassDeclSyntax`, `FunctionDeclSyntax`, or `ProtocolDeclSyntax`.
+- **Implicit**: Represents names introduced by some scopes implicitly, without an associated token representation. Such names include `self`, `newValue`, or `error`.
+
+## Specialized Scopes
+Some scopes share common behavior and can be further generalized as follows:
+- **Type Scope** (`TypeScopeSyntax`): Implicitly introduces `self` and `Self` names.
+- **Sequential Scope** (`SequentialScopeSyntax`): Interleaves its own results with results produced by introducing to sequential parent scopes when encountered during name extraction. Results are ordered from closest to furthest from the unqualified lookup position.
+- **Introducing to Sequential Parent Scope** (`IntroducingToSequentialParentScopeSyntax`): Produces a result that is later interleaved with the results produced by its sequential parent.
+
+> Example of Sequential Scope behavior (code block):
```

Comment:
You could render this as a box in docc if you do 

```
> Example:
> Sequential Scope behavior (code block):
```

Not sure if you’d prefer that but it would give some color to the document.

Reply:
Yeah, I like the idea of distinguishing the code examples a bit more. I added `> Example:` at the start of both of the notes.